### DISCUSSION TITLE
Using `vcpkg` to manage dependencies in a Meson project

### DISCUSSION THREAD

**Discussion by integraledelebesgue** [mesonbuild/meson#13759] [Q&A] [answered]
Hi! How is it possible to locate dependencies installed via `vcpkg` using Meson's `dependency` function?
Can I provide an exact path the headers and libraries are located in?
Also, libraries I use don't provide `.pc` files when downloaded by `vcpkg`, so it's not possible to locate them using `pkg-config`.

**Comment by andy5995** [comment, marked as answer]
@integraledelebesgue You might find what you're looking for in https://github.com/mesonbuild/meson/issues/3500 but as far as I know, if a package doesn't have a .pc file, you would have to use the [compiler.find_library() function](https://mesonbuild.com/Reference-manual_returned_compiler.html#compilerfind_library).

**Reply by integraledelebesgue** [reply]
Thank you for your reply! I managed to link my dependency with `find_library`.
Also, is there a way to discover or pass a vcpkg triplet from the environment in Meson script? I had to hardcode it, which is not good in context of my multiplatform build.

**Reply by andy5995** [reply]
@integraledelebesgue Maybe using a [native file](https://mesonbuild.com/Machine-files.html#cross-and-native-file-reference) might be a good option, along with [get_external_property()](https://mesonbuild.com/Release-notes-for-0-54-0.html#native-file-properties)

**Reply by integraledelebesgue** [reply]
I'll try using that method. Thank you for your help :)
